### PR TITLE
BAU-553: Improve user management - Part 1

### PR DESCRIPTION
**Pull request by jamescutts**
This PR updates the Platform administration functions to better support the BAU team in the management of users.

This specific PR does the following:
- BAU-549 - Improves user invites with ability to assign role
- BAU-550 -Moves "Platform administration" from the footer to the main menu with associated permissions check
- BAU-551 - Adds a"Pre-release" users section in platform administration for display of pre-release users (and thus removing this from the main user management page)
- BAU-552 - Displays a users role on the user management page-
- Updates the Platform administration page to be a bit more user friendly than a text list of links.
- BAU-554 - Add the ability to cancel a user invite

⚠️ Contains some WIP
This PR Also contains a small amount of UI WIP for Part 2 which will add the ability to edit an existing users role.

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/bau/BauUsersPage.tsx [hunk 1]
```diff
@@ -50,6 +52,10 @@ const BauUsersPage = () => {
                   {user.name}
                 </th>
                 <td className="govuk-table__cell">{user.email}</td>
+                <td className="govuk-table__cell">{user.role}</td>
+                <td className="govuk-table__cell">
+                  {/* <Link to={`/administration/users/${user.id}`}>Manage</Link> */}
```
Comment (jamescutts) [inline src/explore-education-statistics-admin/src/pages/bau/BauUsersPage.tsx:57]: Disabled this link as this is a WIP page

File: src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+interface Model {
+  user: UserStatus;
+}
+
+interface FormValues {
+  userEmail: string;
+}
+
+const ManageUserPage = ({
```
Comment (jamescutts) [inline src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx:41]: This page is WIP. Disabled link to access it and currently doesn't perform any actions.

File: src/explore-education-statistics-admin/src/pages/users/PendingInvitesPage.tsx [hunk 3]
```diff
@@ -38,13 +43,25 @@ const PendingInvitesPage = () => {
...
         {model && (
           <tbody className="govuk-table__body">
             {model.users.map(user => (
-              <tr className="govuk-table__row" key={user}>
-                <td className="govuk-table__cell">{user}</td>
+              <tr className="govuk-table__row" key={user.email}>
+                <td className="govuk-table__cell">{user.email}</td>
+                <td className="govuk-table__cell">{user.role}</td>
+                <td className="govuk-table__cell">
+                  {/* <ButtonText onClick={cancelInviteHandler}>
```
Comment (jamescutts) [inline src/explore-education-statistics-admin/src/pages/users/PendingInvitesPage.tsx:61]: Planned functionality

File: src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        {model && (
+          <tbody className="govuk-table__body">
+            {model.users.map(user => (
+              <tr className="govuk-table__row" key={user.id}>
+                <th scope="row" className="govuk-table__header">
+                  {user.name}
+                </th>
+                <td className="govuk-table__cell">{user.email}</td>
+                <td className="govuk-table__cell">
+                  {/* <ButtonText onClick={removeAccessHander}>Remove</ButtonText> */}
```
Comment (jamescutts) [inline src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx:63]: Planned for Part 2

File: src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  useEffect(() => {
+    userService.getPreReleaseUsers().then(users => {
+      setModel({
+        users,
+      });
+    });
+  }, []);
+
+  const removeAccessHander = () => {};
```
Comment (jamescutts) [inline src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx:23]: Planned for Part 2

File: src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  match,
+  history,
+}: RouteComponentProps<{ userId: string }>) => {
+  const { userId } = match.params;
+  const [model, setModel] = useState<Model>();
+  const formId = userId;
+
+  useEffect(() => {
+    userService.getUser(userId).then(user => {
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx:50]: Would suggest using the `useAsyncRetry` hook as it handles loading/error states. Ideally we would want to show a `LoadingSpinner` using it's `isLoading` flag.

File: src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+  }, []);
+
+  const cancelHandler = () => history.push('/administration/users');
+
+  const handleSubmit = useFormSubmit<FormValues>(async values => {
+    // const submission: UserInvite = {
+    //   email: values.userEmail,
+    // };
+
+    // await userService.inviteUser(submission);
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx:64]: We're not doing anything here?

File: src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+interface Model {
+  users: UserStatus[];
+}
+
+const PreReleaseUsersPage = () => {
+  const [model, setModel] = useState<Model>();
+
+  useEffect(() => {
+    userService.getPreReleaseUsers().then(users => {
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx:16]: Same as earlier. Would suggest using the `useAsyncRetry` hook

File: src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx (hunk 7 above)
Comment (jamescutts) [reply src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx:66]: No this page just has its UI mocked out for the purpose of product owner review.

File: src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx (hunk 6 above)
Comment (jamescutts) [reply src/explore-education-statistics-admin/src/pages/users/ManageUserPage.tsx:52]: This page is just a mock at the moment so not too concerned at the moent as this page will be fully implemented by BAU-557 and BAU-558.

File: src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx (hunk 8 above)
Comment (jamescutts) [reply src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx:16]: Have we got any simple examples of this? I had a look but tbh they all looked over my head for my knowledge of react at the moment. Happy for somebody else to have a look but given this page will have limited use dont want this hung up to long.

Comment (jamescutts) [reply src/explore-education-statistics-admin/src/pages/users/PreReleaseUsersPage.tsx]: useAsyncRetry implemented

**Pull request merged**